elastic/kibana
# [Exploratory view] remove add series button tooltip by managing focus

## Summary
Fixes https://github.com/elastic/kibana/issues/119943

For report types that support only 1 series, a tooltip is displayed over the disabled `Add Series` button when a user attempt to add more than 1 series. Previously, the tooltip remained for the duration of the session. 

Tooltips are only supposed to remain as long as the element is focused. Although the user continued to interact with other focusable elements on the page, the tooltip remains. This appears to be due to buggy focus management.

This PR prevents the tooltip from remaining on the `Add Series` button by introducing better focus management. The `Add Series` button is now deliberately blurred when clicked and placed on the next important interactable element, the `data type`  button. Once the data type is selected, the focus is further moved to the `report type` button. Notably, the add series tooltip no longer remains for the duration of the session.

![chrome-capture (30)](https://user-images.githubusercontent.com/11356435/148086922-f0ddd367-d13d-4ee4-968d-1ae4f3ce77b6.gif)

**Testing**
1. Navigate to exploratory view with some established data. The best way to do this is to connect your local version of Kibana to `edge-oblt` using `oblt-cli`.
2. Select `Core Web Vitals` for the report type
3. Add a series using the `Add Series` button
4. Hover over the now disabled `Add Series` button to display the tooltip.
5. Move off the `Add Series` button. Notice how the tooltip disappears.
6. Continue interacting with the page. The tooltip should no longer remain
